docs/admin_design.md

```diff
@@ -0,0 +1,144 @@
++ update category
++ get individual category
++ get all categories
+
+## 2. Model Design
+
+For category, there is 3 special requirements:
+
+* order
+
+    a new category will be the last one in category list. order of category can
+    be changed.
+
+* multilevel
+
+    a category can have one parent category. each root category and its
+    descendents form a tree structure.
+
+* multi-languang
+
+    name, description, meta title, meta description, meta keywords of category
+    should be multiple language.
+
+Following is how to achieve those requirement.
+
+### 2.1 OrderHint Design
+
+In category model, we store a string named `orderHint` and use it to sort
+category in one level. An `orderHint` is a string, the context is the value of 0
+to 1 float data.
+
+When create a category, will create an `orderHint` to it by system time, here is
+the rule:
+
+1. get system current time in milliseconds, like: 14919906819555.
+
+2. convert step 1 value to float data, 14919906819555 -> 0.14919906819555.
+
+3. store setp 2 value to `orderHint` in string.
+
+When update category order, what we need to do is update one category
+`orderHint` and others don't need to change, here is the rule:
+
+1. get the first one category `orderHint` and after one category `orderHint`,
+   calculate the median of two `orderHint`.  example:
+
+   ```
+   the first one orderHint: 0.149199068195555
+   the after one orderHint: 0.149199068195556
+
+   the result: (0.149199068195555 + 0.149199068195556) / 2 = 0.1491990681955555
+   ```
+2. if change the category to the first one, the new `orderHint` will be the
+    median of the original first one `orderHint` and `0`.  example:
+
+    ```
+    the orginal first one orderHint: 0.149199068195555
+
+    the result: (0.149199068195555 + 0) / 2 = 0.074599534097775
+    ```
+
+3. if change the category to the last one, the new `orderHint` will be the
+    median of the original last one `orderHint` and `1`.  exmaple:
+
+    ```
+    the orginal last one orderHint: 0.149199068195555
+
+    the result: (0.149199068195555 + 1) / 2 = 0.574599534097775
+    ```
+
+### 2.2 Multilevel Design
+A category could at most have one parent category. If a category has no parent
+category, this category is a root category. When creating a new category, its
+parent could be added in the meanwhile, here is the rule:
+
+1. get parent id from categoryDraft object from admin-web
+   
+2. set parent and ancestor for this new category. 
+
+3. in setp 2, if its parent id is null, it set its parent as empty(thus, it is a
+   root category), otherwise, just setting parent for this new category.Be
+   careful, if the parent of this new category is not a root category, the new
+   category will have grandparent category, its parent and grandparent should be
+   set as ancestor. For example: if you want to create new category named `cpu`,
+   then its parent category is `computer`, and `computer` has a parent category
+   named `electronic product`
+
```

Comment:
1. missing ancestors definition. ancestors is a list of string , store the path from root category to parent category.
2. step 3 is for what？sub step of step 2? 
3. what heppen if change parent? should change it's ancestors? should change sub category's ancestors?